python/samples/getting_started_with_processes/step01_processes.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+from typing import Generic, TypeVar
+
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.open_ai import OpenAIChatCompletion
+from semantic_kernel.contents import ChatHistory
+from semantic_kernel.functions import kernel_function
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.processes.kernel_process.kernel_process_step_context import KernelProcessStepContext
+from semantic_kernel.processes.kernel_process.kernel_process_step_state import KernelProcessStepState
+from semantic_kernel.processes.local_runtime.local_event import KernelProcessEvent
+from semantic_kernel.processes.process_builder import ProcessBuilder
+from semantic_kernel.processes.process_function_target_builder import ProcessFunctionTargetBuilder
+
+
+class CommonEvents:
+    UserInputReceived = "UserInputReceived"
+    AssistantResponseGenerated = "AssistantResponseGenerated"
+
+
+class ChatBotEvents:
+    StartProcess = "startProcess"
+    IntroComplete = "introComplete"
+    AssistantResponseGenerated = "assistantResponseGenerated"
+    Exit = "exit"
+
+
+class UserInputState(KernelBaseModel):
+    user_inputs: list[str] = []
+    current_input_index: int = 0
+
+
+# Create a type variable for the generic state
+TState = TypeVar("TState", bound=KernelBaseModel)
+
+
+class KernelProcessStep(Generic[TState]):
```

Comment:
I need to clean this up... this exists in the framework and doesn't need to be included in this sample.